main/Main.hs

```diff
@@ -1,398 +1,36 @@
-      return SCLoaded
-
-{-|
-  Starts a dedicated pg connection to LISTEN for notifications.
-  When a NOTIFY <db-channel> - with an empty payload - is done, it refills the schema cache.
-  It uses the connectionWorker in case the LISTEN connection dies.
--}
-listener :: ByteString -> Text -> P.Pool -> IORef AppConfig -> IORef (Maybe DbStructure) -> MVar ConnectionStatus -> IO () -> IO ()
-listener dbUri dbChannel pool refConf refDbStructure mvarConnectionStatus connWorker = start
-  where
-    start = do
-      connStatus <- takeMVar mvarConnectionStatus -- takeMVar makes the thread wait if the MVar is empty(until there's a connection).
-      case connStatus of
-        Connected actualPgVersion -> void $ forkFinally (do -- forkFinally allows to detect if the thread dies
-          dbOrError <- C.acquire dbUri
-          -- Debounce in case too many NOTIFYs arrive. Could happen on a migration(assuming a pg EVENT TRIGGER is set up).
-          -- This might not be needed according to pg docs https://www.postgresql.org/docs/12/sql-notify.html:
-          -- "If the same channel name is signaled multiple times from the same transaction with identical payload strings, the database server can decide to deliver a single notification only."
-          -- But we do it to be extra safe.
-          scFiller <- mkDebounce (defaultDebounceSettings {
-                        -- It's not necessary to check the loadSchemaCache success here. If the connection drops, the thread will die and proceed to recover below.
-                        debounceAction = void $ loadSchemaCache pool actualPgVersion refConf refDbStructure,
-                        debounceEdge = trailingEdge, -- wait until the function hasn’t been called in _1s
-                        debounceFreq = _1s })
-          case dbOrError of
-            Right db -> do
-              putStrLn $ "Listening for notifications on the " <> dbChannel <> " channel"
-              let channelToListen = N.toPgIdentifier dbChannel
-              N.listen db channelToListen
-              N.waitForNotifications (\_ msg ->
-                if BS.null msg
-                  then scFiller    -- reload the schema cache
-                  else pure ()) db -- Do nothing if anything else than an empty message is sent
-            _ -> die errorMessage)
-          (\_ -> do -- if the thread dies, we try to recover
-            putStrLn retryMessage
-            connWorker -- assume the pool connection was also lost, call the connection worker
-            start)     -- retry the listener
-        _ ->
-          putStrLn errorMessage -- Should be unreachable. connectionStatus will retry until there's a connection.
-    errorMessage = "Could not listen for notifications on the " <> dbChannel <> " channel" :: Text
-    retryMessage = "Retrying listening for notifications on the " <> dbChannel <> " channel.." :: Text
+main :: IO ()
+main = CLI.app installHandlers runWithSocket
```

Comment:
That will not quite work, as we need to swallow the unused arguments... I had a earlier version with lambdas, which was messier. An alternative would be passing Maybes into app, but that would cause a bit of more complexity there. WDYT?

Reply:
I see. Using Maybe here sounds semantically correct.

Reply:
Fixed!